modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ConversionUtils.scala

```diff
@@ -477,4 +483,43 @@ object ConversionUtils {
   /** Extract valid IP (v4 or v6) address from a string */
   def extractInetAddress(arg: String): Option[InetAddress] =
     Option(new HostName(arg).asInetAddress)
+
+  private val eeTstampFormatter = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSS")
+
+  /** Creates a PII event from the pii field of an existing event. */
+  def getPiiEvent(processor: Processor, event: EnrichedEvent): Option[EnrichedEvent] =
+    Option(event.pii)
+      .filter(_.nonEmpty)
+      .map { pii =>
+        val ee = new EnrichedEvent
+        ee.unstruct_event = pii
+        ee.app_id = event.app_id
+        ee.platform = "srv"
+        ee.etl_tstamp = event.etl_tstamp
+        ee.collector_tstamp = event.collector_tstamp
+        ee.event = "pii_transformation"
+        ee.event_id = UUID.randomUUID().toString
+        ee.derived_tstamp = eeTstampFormatter.print(DateTime.now(DateTimeZone.UTC))
+        ee.true_tstamp = ee.derived_tstamp
+        ee.event_vendor = "com.snowplowanalytics.snowplow"
+        ee.event_format = "jsonschema"
+        ee.event_name = "pii_transformation"
+        ee.event_version = "1-0-0"
+        ee.v_etl = MiscEnrichments.etlVersion(processor)
+        ee.contexts = getContextParentEvent(ee.event_id).noSpaces
+        ee
+      }
+
+  private def getContextParentEvent(eventId: String): Json =
+    Json.obj(
+      "schema" := Json.fromString("iglu:com.snowplowanalytics.snowplow/contexts/jsonschema/1-0-0"),
```

Comment:
I'd probably use `SelfDescribingData` with serialization here. Just in sake of having `SchemaKey("com.snowplowanalytics.snowplow", "parent_event", "jsonschema", SchemaVer.Full(1,0,0)` constant defined somewhere. But it's very stylistic.